### PR TITLE
update C++ STD from c++11 to c++14

### PR DESCRIPTION
**Pull request by dagar**
- temporarily disable snapdragon builds until toolchain is updated

### CODE REVIEW

**Comment by dagar** [comment]
Overall the change is quite minimal. 

Screenshot from 2020-01-15 12-04-18

File: platforms/nuttx/cmake/px4_impl_os.cmake [hunk 1]
```diff
@@ -67,6 +67,8 @@ function(px4_os_add_flags)
 	# prevent using the toolchain's std c++ library
 	add_compile_options($<$<COMPILE_LANGUAGE:CXX>:-nostdinc++>)
 
+	add_compile_options($<$<COMPILE_LANGUAGE:CXX>:-fno-sized-deallocation>)
```
Comment (bkueng) [inline platforms/nuttx/cmake/px4_impl_os.cmake]: Why do we need that here as well?

File: src/drivers/uavcan/uavcan_drivers/kinetis/driver/src/uc_kinetis_flexcan.cpp [hunk 2]
```diff
@@ -693,7 +693,7 @@ void CanIface::handleRxInterrupt(uavcan::uint32_t rx_iflags, uavcan::uint64_t ut
 		frame.data[7] = rf.data.b7;
 
 		volatile uavcan::uint32_t idhit = can_->RXFIR;
-		UNUSED(idhit);
+		//UNUSED(idhit);
```
Comment (bkueng) [inline src/drivers/uavcan/uavcan_drivers/kinetis/driver/src/uc_kinetis_flexcan.cpp]: Why not just remove this?

File: platforms/nuttx/cmake/px4_impl_os.cmake (hunk 1 above)
Comment (dagar) [reply platforms/nuttx/cmake/px4_impl_os.cmake]: The flags set by `add_compile_options` aren't picked up by the NuttX build.

File: src/drivers/uavcan/uavcan_drivers/kinetis/driver/src/uc_kinetis_flexcan.cpp (hunk 2 above)
Comment (dagar) [reply src/drivers/uavcan/uavcan_drivers/kinetis/driver/src/uc_kinetis_flexcan.cpp]: No reason, just wasn't treating it like "our" code.
It was also kind of a peculiar error.

Removed.

**Comment by dagar** [comment]
2014 here we come...

File: msg/templates/urtps/microRTPS_agent_CMakeLists.txt.em [hunk 3]
```diff
@@ -38,15 +38,15 @@ project(micrortps_agent)
...
 find_package(fastcdr REQUIRED)
 
-# Set C++11
+# Set C++14
 include(CheckCXXCompilerFlag)
 if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_COMPILER_IS_CLANG OR
     CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-    check_cxx_compiler_flag(--std=c++11 SUPPORTS_CXX11)
+    check_cxx_compiler_flag(--std=c++14 SUPPORTS_CXX14)
     if(SUPPORTS_CXX11)
```
Comment (tzai) [inline msg/templates/urtps/microRTPS_agent_CMakeLists.txt.em]: @dagar shouldn't have this been changed to `SUPPORTS_CXX14` as well?

Comment (dagar) [reply msg/templates/urtps/microRTPS_agent_CMakeLists.txt.em]: Yes I suppose so, although the agent is actually a separate build. Care to open a PR fixing it?

Good catch.

Comment (tzai) [reply msg/templates/urtps/microRTPS_agent_CMakeLists.txt.em]: Sure

**Pull request merged**